src/NuGet.Core/NuGet.Commands/PackagesLockFileBuilder.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using NuGet.Common;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Shared;
+
+namespace NuGet.Commands
+{
+    public class PackagesLockFileBuilder
+    {
+        private readonly int _lockFileVersion;
+
+        public PackagesLockFileBuilder(int lockFileVersion)
+        {
+            _lockFileVersion = lockFileVersion;
+        }
+
+        public PackagesLockFile CreateNuGetLockFile(LockFile assetsFile)
+        {
+            var lockFile = new PackagesLockFile()
+            {
+                Version = _lockFileVersion
+            };
+
+            var libraryLookup = assetsFile.Libraries.Where(e => e.Type == LibraryType.Package)
+                .ToDictionary(e => new PackageIdentity(e.Name, e.Version));
```

Comment:
why do you need a dictionary if there is no key value pair? convert the IEnumerable to a HashSet instead.

Reply:
Look closely, that is actually a dictionary where `key` is `PackageIdentity` and `value` is `LockFIleLibrary`